src/customActions/index.js

```diff
@@ -103,4 +103,11 @@ export default (manager) => {
     
   })
 
+  //Customize the Outbound Call Action
+  Actions.replaceAction("StartOutboundCall", (payload, original) => {
+    const state = manager.store.getState();
+    var newPayload = payload;
+    newPayload.callerId = state.dialpad.outgoingNumber.selectedOutgoingNumber?.phoneNumber;
```

Comment:
Need to clear the value once you have used it, as discussed, you only want to use this number once. Note: from reading through the code I don't think this currently happens but might be worth a quick test first just to make sure I didn't miss anything :-).